tests/codegen/issue-103840.rs

```diff
@@ -1,4 +1,8 @@
+// should-fail
+// known-bug: #103840
 // compile-flags: -O
+// FIXME: reenable after next llvm update. currently the codegen is too fickle, minor layout
+// perturbations in Vec break this test.
```

Comment:
see also #106790 and nikic's comment there https://github.com/rust-lang/rust/pull/106790#issuecomment-1382904714